homeassistant/components/tesla_wall_connector/sensor.py

```diff
@@ -0,0 +1,141 @@
+        value_getter=lambda data: data[WALLCONNECTOR_DATA_VITALS].grid_v,
+        device_class=DEVICE_CLASS_VOLTAGE,
+        state_class=STATE_CLASS_MEASUREMENT,
+        entity_category=ENTITY_CATEGORY_DIAGNOSTIC,
+    ),
+    WallConnectorSensorDescription(
+        key="grid_hz",
+        name=prefix_entity_name("Grid Frequency"),
+        native_unit_of_measurement=FREQUENCY_HERTZ,
+        value_getter=lambda data: data[WALLCONNECTOR_DATA_VITALS].grid_hz,
+        state_class=STATE_CLASS_MEASUREMENT,
+        entity_category=ENTITY_CATEGORY_DIAGNOSTIC,
+    ),
+    WallConnectorSensorDescription(
+        key="power",
+        name=prefix_entity_name("Power"),
+        native_unit_of_measurement=POWER_KILO_WATT,
+        value_getter=lambda data: round(
+            (
+                (
+                    data[WALLCONNECTOR_DATA_VITALS].currentA_a
+                    * data[WALLCONNECTOR_DATA_VITALS].voltageA_v
+                )
+                + (
+                    data[WALLCONNECTOR_DATA_VITALS].currentB_a
+                    * data[WALLCONNECTOR_DATA_VITALS].voltageB_v
+                )
+                + (
+                    data[WALLCONNECTOR_DATA_VITALS].currentC_a
+                    * data[WALLCONNECTOR_DATA_VITALS].voltageC_v
+                )
+            )
+            / 1000.0,
+            1,
+        ),
+        device_class=DEVICE_CLASS_POWER,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    WallConnectorSensorDescription(
+        key="total_energy_kWh",
+        name=prefix_entity_name("Total Energy"),
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        value_getter=lambda data: data[WALLCONNECTOR_DATA_LIFETIME].energy_wh / 1000.0,
+        state_class=STATE_CLASS_TOTAL_INCREASING,
+        device_class=DEVICE_CLASS_ENERGY,
+    ),
+]
+
+
+async def async_setup_entry(hass, config_entry, async_add_devices):
+    """Create the Wall Connector sensor devices."""
+    wall_connector = hass.data[DOMAIN][config_entry.entry_id]
+
+    all_entities = []
+    for description in wall_connector_sensors:
+        entity = WallConnectorSensorEntity(wall_connector, description)
+        if entity is not None:
+            all_entities.append(entity)
+
+    async_add_devices(all_entities)
+
+
+class WallConnectorSensorEntity(WallConnectorEntity, SensorEntity):
+    """Wall Connector Sensor Entity."""
+
+    entity_description: WallConnectorSensorDescription
+
+    def __init__(
+        self, wall_connector: dict, description: WallConnectorSensorDescription
+    ) -> None:
+        """Initialize WallConnectorSensorEntity."""
+        self.entity_description = description
+        super().__init__(wall_connector)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if self.coordinator.data is None:
+            return None
+
+        return self.entity_description.value_getter(self.coordinator.data)
```

Comment:
```suggestion
```

Please limit initial contributions to a single platform.

https://developers.home-assistant.io/docs/creating_component_code_review/#5-make-your-pull-request-as-small-as-possible

Reply:
Removing the sensor platform from this integration would make it pretty useless since it wouldn't contain the energy sensor, which for me was the reason to integrate the Tesla Wall Connector into Home Assistant in order to have it shown up in the energy dashboard. 

I could remove the binary_sensor integration but that would mean that the users wouldn't be able to create an automation on when the car is plugged in / charging. I understand that pull requests should be kept small but I find it a bit pointless to limit it to just one platform, the binary_sensor platform doesn't add much code. If I leave it it, would it just be in order to make a new pull request as soon as this one is merged?

Reply:
Once the initial merge is done, open additional PRs to add the other platforms.  Usually we want the smallest platform first since it keeps the initial PR small since large platforms in initial
PRs can take months to find a reviewer willing to work through it with the OP

Reply:
Sensors removed in 5e57560